README.md

```diff
@@ -56,14 +56,45 @@ expect(response.body).to match_snapshot("api/resources_index")
 
 # Using plain text instead of parsing JSON
 expect(response.body).to match_snapshot("api/resources_index", format: :text)
+```
+
+#### Matcher options
+
+##### dynamic_attributes
+
+Using `dynamic_attributes` inline allows to ignore attributes for a specific snapshot.
+This is useful to ignore changing attributes, like `id` or `created_at`.
+Notice that **all** nodes matching those (nested or not) will be ignored. Usage:
 
+```ruby
 # Defining specific test dynamic attributes
 expect(response.body).to match_snapshot("api/resources_index", dynamic_attributes: %w(confirmed_at relation_id))
+```
+
+##### ignore_order
+
+It is possible to use the `ignore_order` inline option to mark which array nodes are unsorted and that elements position
+should not be taken into consideration.
 
+```ruby
 # Ignoring order for certain arrays (this will ignore the ordering for the countries array inside the json response)
 expect(response.body).to match_snapshot("api/resources_index", ignore_order: %w(countries))
 ```
 
+**Note:** `ignore_order` has some limitations:
+
+- If you are using it for arrays of objects/hashes (ie: `[{name: "name", value: "value"}, ...]`),
+it won't perform well depending on the array and hash size (number of keys)
+
+- Due to limitations on sorting array of objects/hashes, `ignore_order` might fail for cases where the array elements
+don't have matching keys. For example, it is hard to sort the following arrays in a consistent way
+(even tho they have the same elements):
+
+```
+[{z: 2, a: 2}, {b: 5, z: 1}, {a: 2, b: 5}, {c: 10, a: 2, b: 3}, {d: 4, a: 1}, {d: 6, b: 1}, {a: [1, 3]}, {a: {a: 1}}]
```

Comment:
To solve this one specifically just do this:

```
array1 = [{z: 2, a: 2}, {b: 5, z: 1}, {a: 2, b: 5}, {c: 10, a: 2, b: 3}, {d: 4, a: 1}, {d: 6, b: 1}, {a: [1, 3]}, {a: {a: 1}}]
array2 = [{a: {a: 1}}, {z: 2, a: 2}, {b: 5, z: 1}, {d: 6, b: 1}, {a: 2, b: 5}, {c: 10, a: 2, b: 3}, {a: [1, 3]}, {d: 4, a: 1}]
diff = (array1 - array2) + (array2 - array1)
equal = diff.empty?
```

```
irb(main):014:0> equal = diff.empty?
=> true

```

Reply:
@danielnc yes, I was working with a piece of code like this prior to the current solution. To do this we would need to do node by node comparison again. And then it gets more complex than this, because you posted how to check if 2 hash arrays match (not how to sort). 

If we have the following case:

```
# snapshot
data: [
  node1: {
    a: [
      {b: 2, c: [1, 2, 3], d: 8, e: [7, 8, 9]},
      {b: 2, c: "A", d: 9, e: [11, 12]},
      {b: 2, c: [4], d: 9, e: [11, 12]},
      {b: 2, c: 1, d: 9, e: [11, 12]},
    ]
  }
  node2: {
    a: [1, 2, 3]
  }
]

# comparison json
data: [
  node1: {
    a: [
      {b: 2, c: "A", d: 9, e: [11, 12]},
      {b: 2, c: [3, 1, 2], d: 8, e: [7, 8, 9]},
      {b: 2, c: [4], d: 9, e: [11, 12]},
      {b: 2, c: 1, d: 9, e: [11, 12]},
    ]
  }
  node2: {
    a: [3, 1, 2]
  }
]
```

Say `ignore_order` is `%w(a c)`. We can't just do `(snapshot_node1[a] - comparison_node1[a]) == (comparison_node1[a] - snapshot_node1[a])` because it will fail, due to `c` not having the same order. And the more you start nesting it, the harder it becomes to deal with it. I will remove this limitation text. If someone ever encounters this limitation, I can explain it works like that.